lldb/tools/lldb-dap/Protocol/ProtocolTypes.cpp

```diff
@@ -44,4 +43,190 @@ bool fromJSON(const json::Value &Params, Source &S, json::Path P) {
          O.mapOptional("sourceReference", S.sourceReference);
 }
 
+json::Value toJSON(const ExceptionBreakpointsFilter &EBF) {
+  json::Object result{{"filter", EBF.filter}, {"label", EBF.label}};
+
+  if (EBF.description)
+    result.insert({"description", *EBF.description});
+  if (EBF.defaultState)
+    result.insert({"default", *EBF.defaultState});
+  if (EBF.supportsCondition)
+    result.insert({"supportsCondition", *EBF.supportsCondition});
+  if (EBF.conditionDescription)
+    result.insert({"conditionDescription", *EBF.conditionDescription});
+
+  return result;
+}
+
+json::Value toJSON(const ColumnType &T) {
+  switch (T) {
+  case eColumnTypeString:
+    return "string";
+  case eColumnTypeNumber:
+    return "number";
+  case eColumnTypeBoolean:
+    return "boolean";
+  case eColumnTypeTimestamp:
+    return "unixTimestampUTC";
+  }
+}
+
+json::Value toJSON(const ColumnDescriptor &CD) {
+  json::Object result{{"attributeName", CD.attributeName}, {"label", CD.label}};
+
+  if (CD.format)
+    result.insert({"format", *CD.format});
+  if (CD.type)
+    result.insert({"type", *CD.type});
+  if (CD.width)
+    result.insert({"width", *CD.width});
+
+  return result;
+}
+
+json::Value toJSON(const ChecksumAlgorithm &CA) {
+  switch (CA) {
+  case eChecksumAlgorithmMD5:
+    return "MD5";
+  case eChecksumAlgorithmSHA1:
+    return "SHA1";
+  case eChecksumAlgorithmSHA256:
+    return "SHA256";
+  case eChecksumAlgorithmTimestamp:
+    return "timestamp";
+  }
+}
+
+json::Value toJSON(const BreakpointModeApplicability &BMA) {
+  switch (BMA) {
+  case eBreakpointModeApplicabilitySource:
+    return "source";
+  case eBreakpointModeApplicabilityException:
+    return "exception";
+  case eBreakpointModeApplicabilityData:
+    return "data";
+  case eBreakpointModeApplicabilityInstruction:
+    return "instruction";
+  }
+}
+
+json::Value toJSON(const BreakpointMode &BM) {
+  json::Object result{
+      {"mode", BM.mode},
+      {"label", BM.label},
+      {"appliesTo", BM.appliesTo},
+  };
+
+  if (BM.description)
+    result.insert({"description", *BM.description});
+
+  return result;
+}
+
+static llvm::StringLiteral ToString(AdapterFeature feature) {
+  switch (feature) {
+  case eAdapterFeatureSupportsANSIStyling:
+    return "supportsANSIStyling";
+  case eAdapterFeatureSupportsBreakpointLocationsRequest:
+    return "supportsBreakpointLocationsRequest";
```

Comment:
I feel bad for not thinking about this earlier, but now that I see this switch, I think this would' bea good candidate for a `.def` file. We could have a `dap.def` file that defines and use that to implement methods for both serializing and deserializing. That doesn't need to be part of this PR though.